crates/noirc_evaluator/src/ssa/code_gen.rs

```diff
@@ -86,10 +95,14 @@ impl<'a> IRGenerator<'a> {
     }
 
     pub fn find_variable(&self, variable_def: DefinitionId) -> Option<&Value> {
-        self.variable_values.get(&variable_def)
+        if variable_def != DefinitionId::dummy_id() {
+            self.variable_values.get(&variable_def)
```

Comment:
No, there was not a bug, I added this because my initial implementation was using dummy DefinitionId for member variables arguments that was missing it. I am not doing this finally but I kept this because I think it is a good sanity check in any case.